meadServer/client/src/App.js

```diff
@@ -1,3 +1,4 @@
+
```

Comment:
```
     this.setState({ selected: newSelection });
     this.getData();
```

I don't think this is guaranteed to work. setState isn't synchronous. setState takes a callback function as a parameter. Alternatively, in `componentDidUpdate` or whichever applicable lifecycle function, you can check if the data loaded is equal to the selected label, and if not update the data.

Also consider what would happen if the server was slow to respond. Could you show some sort of UX while the data is loading so it's clear to the user the displayed graph is not the right one?